docker/synapse.Dockerfile

```diff
@@ -21,10 +18,13 @@ RUN mkdir /rust /cargo
 
 RUN curl -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --default-toolchain stable --profile minimal
 
-# For now, we need to tell Debian we don't care that we're editing the system python
-# installation.
-# Some context in https://github.com/pypa/pip/issues/11381#issuecomment-1399263627
-RUN ${PYTHON_VERSION} -m pip install -q --no-cache-dir poetry==1.3.2 ${SYSTEM_PIP_INSTALL_SUFFIX}
+ARG PYTHON_VERSION=3.9
```

Comment:
I'm thinking we should maybe add a comment about this default being the minimum Python version supported according to the [Synapse deprecation policy](https://element-hq.github.io/synapse/latest/deprecation_policy.html).

Especially since this is documented in another project.